stock_inventory/models/stock_inventory.py

```diff
@@ -191,24 +191,56 @@ def _get_domain_category_quants(self, base_domain):
             ]
         )
 
+    def _get_products(self):
+        self.ensure_one()
+        if self.product_selection == "all":
+            return self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            return self.env["product.product"].search(
+                [("categ_id", "=", self.category_id.id)]
+            )
+        else:
+            return self.product_ids
+
+    def _get_overlapping_inventories(self):
+        inventories = self.search([("state", "=", "in_progress")])
+        if not inventories:
+            return self.env["stock.inventory"]
+        for rec in self:
+            inventory_products = rec._get_products()
+            overlapping_inventories = inventories.filtered(
+                lambda x: (
+                    x.id != rec.id
+                    and (
+                        any(i in x.location_ids for i in rec.location_ids)
+                        or (
+                            any(
+                                i in x.location_ids.child_internal_location_ids
+                                for i in rec.location_ids
+                            )
+                            and not x.exclude_sublocation
+                        )
+                    )
+                    and any(
+                        product in inventory_products for product in x._get_products()
+                    )
+                )
+            )
+            return overlapping_inventories
+
     def refresh_stock_quant_ids(self):
         for rec in self:
             rec.stock_quant_ids = rec._get_quants(rec.location_ids)
 
     def action_state_to_in_progress(self):
-        active_rec = self.env["stock.inventory"].search(
-            [
-                ("state", "=", "in_progress"),
-                ("location_ids", "child_of", self.location_ids.ids),
-            ],
-            limit=1,
-        )
-        if active_rec:
+        overlapping_inventories = self._get_overlapping_inventories()
+        if overlapping_inventories:
             raise ValidationError(
                 _(
-                    "There's already an Adjustment in Process using one requested Location: %s"
+                    "There's already an Adjustment in Process using one requested Location"
+                    " and Product: %s"
                 )
-                % active_rec.name
+                % overlapping_inventories.name
```

Comment:
Using translation function arguments is better.

```suggestion
                _(
                    "There's already an Adjustment in Process using one requested Location"
                    " and Product: %(name)s", overlapping_inventories.name
                    )
```

Reply:
Sure!, thanks